src/wallet/scriptpubkeyman.cpp

```diff
@@ -842,7 +895,14 @@ void LegacyScriptPubKeyMan::SetHDChain(const CHDChain& chain, bool memonly)
     if (!memonly && !WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain))
```

Comment:
As per the review club discussion, the intended behaviour is only effective when the wallet is reloaded. As `SetHdChain` is writing the chains in the disk, `CWalletScanState` can only observe the presence of new chain if it reads back from the disk. This means after setting the seed user must reload the wallet to track inactive chains which might not be readily obvious. `SetHdChain` should detect that a previous chain is being replaced and correspondingly call  `AddInactiveHDChain` before writing the new one to the disk.

Reply:
See https://github.com/bitcoin/bitcoin/pull/17681#discussion_r426066015